### ISSUE TITLE
Terms Aggregation return wrong terms if terms contains dashes "-"

### ISSUE THREAD

**Issue by mponton** [elastic/elasticsearch#16706]
I'm having an issue using ElasticSearch v2.2.0 on Windows.

**Given the following index data:**

```
PS D:\> (Invoke-WebRequest -Uri "http://localhost:9200/pingbeat-2016.02.16/_search?size=10000&pretty=true").Content.Split("`n")
|Select-String target_name |Sort -Unique

        "target_name" : "hwd-u2b-01f.domain.tld",
        "target_name" : "hwd-u2b-02f.domain.tld",
        "target_name" : "hwd-u2b-03f.domain.tld",
        "target_name" : "pon-u2b-01f.domain.tld",
        "target_name" : "pon-u2b-02f.domain.tld",
        "target_name" : "pon-u2b-03f.domain.tld",
        "target_name" : "timeserv.domain.tld",
```

**The following terms aggregation:**

```
{
"size": 0,
"aggs" : {
    "targets" : {
        "terms" : { "field" : "target_name" }
    }
}}
```

**Returns wrong values for targets:**

```
PS D:\> (Invoke-WebRequest -Uri "http://localhost:9200/pingbeat-2016.02.16/_search?pretty=true" -Method POST -InFile .\pb-targets.txt).Content
{
  "took" : 3,
  "timed_out" : false,
  "_shards" : {
    "total" : 5,
    "successful" : 5,
    "failed" : 0
  },
  "hits" : {
    "total" : 26712,
    "max_score" : 0.0,
    "hits" : [ ]
  },
  "aggregations" : {
    "targets" : {
      "doc_count_error_upper_bound" : 0,
      "sum_other_doc_count" : 0,
      "buckets" : [ {
        "key" : "u2b",
        "doc_count" : 22896
      }, {
        "key" : "hwd",
        "doc_count" : 11448
      }, {
        "key" : "pon",
        "doc_count" : 11448
      }, {
        "key" : "01f.domain.tld",
        "doc_count" : 7632
      }, {
        "key" : "02f.domain.tld",
        "doc_count" : 7632
      }, {
        "key" : "03f.domain.tld",
        "doc_count" : 7632
      }, {
        "key" : "timeserv.domain.tld",
        "doc_count" : 3816
      } ]
    }
  }
}
```

As you can see, the aggregation returns the following terms:

* u2b
* hwd
* pon
* 01f.domain.tld
* 02f.domain.tld
* 03f.domain.tld
* timeserv.domain.tld

Please let me know if you need more details to help resolve this issue.

Thanks!

**Comment by xavierfacq** [comment]
Hi,

Your field must be "not_analyzed" if you want to aggregate unique values, use a mapping with something like : 

```
	          "target_name" : {
	            "type" : "string",
                "index": "not_analyzed",
                "include_in_all": false
	          },

```

Else it aggregates distinct terms.

Xavier

**Comment by mponton** [comment]
My apologies.

I'm just getting started with ES and apparently I need to read quite a bit more. I used a default index template from https://github.com/joshuar/pingbeat and indeed, the `target_name` is `analyzed`. I reindexed and all is good now.

I will open a ticket or send Joshua a pull request as I don't see the benefit in having a hostname analyzed. It does not look very good on a dashboard :-)

A huge thank you for replying so quickly and helping me with this!

Cheers,

Marco
